File: ABCRM/WEB/CrmWeb/KFPT/HYHDDY/KFPT_HYHDDY.js
vUrl = "../KFPT.ashx";

$(document).ready(function () {


});


function SetControlState() {
    $("#jlbh").hide();
    $("#B_Stop").show();
    var bHasData = $("#TB_JLBH").val() != "";//有数据
    var bExecuted = $("#LB_ZXRMC").text() != "";//已审核
    document.getElementById("B_Stop").disabled = !(bHasData && bExecuted);
    if ($("#LB_ZXRMC").text() != "") {
        $("#ZZR").show();
        $("#ZZSJ").show();
    }
    if ($("#LB_ZZRMC").text() != "") {
        document.getElementById("B_Stop").disabled = true;

    }


}

function IsValidData() {

    if ($("#TB_RS").val() == "") {
        ShowMessage("预计人数不能为空");
        return false;
    }
    if ($("#TB_HDMC").val() == "") {
        ShowMessage("活动名称不能为空");
        return false;
    }
    if ($("#TB_KSSJ").val() == "") {
        ShowMessage("活动开始时间不能为空");
        return false;
    }
    if ($("#TB_JSSJ").val() == "") {
        ShowMessage("活动结束时间不能为空");
        return false;
    }
    if ($("#TB_KSSJ").val() > $("#TB_JSSJ").val()) {
        ShowMessage("开始时间不能大于结束时间");
        return false;
    }
    if ($("#TB_BM_RQ1").val() == "") {
        ShowMessage("报名开始时间不能为空");
        return false;
    }
    if ($("#TB_BM_RQ2").val() == "") {
        ShowMessage("报名结束时间不能为空");
        return false;
    }
    if ($("#TB_BM_RQ1").val() > $("#TB_BM_RQ2").val()) {
        ShowMessage("报名开始时间不能大于报名结束时间");
        return false;
    }

    //if ($("#TB_QR_RQ1").val() == "") {
    //    art.dialog({ content: "确认开始时间不能为空", lock: true, time: 2 });
    //    return false;
    //}
    //if ($("#TB_QR_RQ2").val() == "") {
    //    art.dialog({ content: "确认结束时间不能为空", lock: true, time: 2 });
    //    return false;
    //}
    //if ($("#TB_QR_RQ1").val() > $("#TB_QR_RQ2").val()) {
    //    art.dialog({ content: "确认开始时间不能大于确认结束时间", lock: true, time: 2 });
    //    return false;
    //}
    if ($("#TB_CJ_RQ1").val() == "") {
        ShowMessage("参加开始时间不能为空");
        return false;
    }
    if ($("#TB_CJ_RQ2").val() == "") {
        ShowMessage("参加结束时间不能为空");
        return false;
    }
    if ($("#TB_CJ_RQ1").val() > $("#TB_CJ_RQ2").val()) {
        ShowMessage("参加开始时间不能大于参加结束时间");
        return false;
    }

    if ($("#TB_HF_RQ1").val() == "") {
        ShowMessage("恢复开始时间不能为空");
        return false;
    }
    if ($("#TB_HF_RQ2").val() == "") {
        ShowMessage("恢复结束时间不能为空");
        return false;
    }
    if ($("#TB_HF_RQ1").val() > $("#TB_HF_RQ2").val()) {
        ShowMessage("恢复开始时间不能大于回复结束时间");
        return false;
    }
    if ($("#TB_PS_RQ1").val() == "") {
        ShowMessage("评述开始时间不能为空");
        return false;
    }
    if ($("#TB_PS_RQ2").val() == "") {
        ShowMessage("评述结束时间不能为空");
        return false;
    }
    if ($("#TB_PS_RQ1").val() > $("#TB_PS_RQ2").val()) {
        ShowMessage("评述开始时间不能大于评述结束时间");
        return false;
    }

    if ($("#TB_BM_RQ1").val() < $("#TB_KSSJ").val() || $("#TB_BM_RQ2").val() > $("#TB_JSSJ").val()) {
        ShowMessage("报名时间必须在活动时间范围之内");
        return false;
    }

    //if ($("#TB_QR_RQ1").val() < $("#TB_KSSJ").val() || $("#TB_QR_RQ2").val() > $("#TB_JSSJ").val()) {
    //    art.dialog({ content: "确认时间必须在活动时间范围之内", lock: true, time: 2 });
    //    return false;
    //}


    if ($("#TB_CJ_RQ1").val() < $("#TB_KSSJ").val() || $("#TB_CJ_RQ2").val() > $("#TB_JSSJ").val()) {
        ShowMessage("参加时间必须在活动时间范围之内");
        return false;
    }


    if ($("#TB_HF_RQ1").val() < $("#TB_KSSJ").val() || $("#TB_HF_RQ2").val() > $("#TB_JSSJ").val()) {
        ShowMessage("回访时间必须在活动时间范围之内");
        return false;
    }
    if ($("#TB_PS_RQ1").val() < $("#TB_KSSJ").val()) {
        ShowMessage("评述开始时间不能小于活动开始时间");
        return false;
    }

    return true;
}

function SaveData() {
    var Obj = new Object();
    Obj.iJLBH = $("#TB_JLBH").val();
    if (Obj.iJLBH == "")
        Obj.iJLBH = "0";

    Obj.sHDMC = $("#TB_HDMC").val();
    Obj.sHDNR = $("#TB_HDNR").val();
    Obj.iRS = $("#TB_RS").val();

    Obj.dKSSJ = $("#TB_KSSJ").val();
    Obj.dJSSJ = $("#TB_JSSJ").val();
    Obj.dBM_RQ1 = $("#TB_BM_RQ1").val();
    Obj.dBM_RQ2 = $("#TB_BM_RQ2").val();
    Obj.dQR_RQ1 = $("#TB_QR_RQ1").val();
    Obj.dQR_RQ2 = $("#TB_QR_RQ2").val();
    Obj.dCJ_RQ1 = $("#TB_CJ_RQ1").val();
    Obj.dCJ_RQ2 = $("#TB_CJ_RQ2").val();
    Obj.dHF_RQ1 = $("#TB_HF_RQ1").val();
    Obj.dHF_RQ2 = $("#TB_HF_RQ2").val();
    Obj.dPS_RQ1 = $("#TB_PS_RQ1").val();
    Obj.dPS_RQ2 = $("#TB_PS_RQ2").val();
    Obj.iLoginRYID = iDJR;
    Obj.sLoginRYMC = sDJRMC;
    return Obj;
}
function ShowData(data) {
    var Obj = JSON.parse(data);
    $("#TB_JLBH").val(Obj.iJLBH);

    $("#TB_HDMC").val(Obj.sHDMC);
    $("#TB_HDNR").val(Obj.sHDNR);
    $("#TB_RS").val(Obj.iRS);
    $("#TB_KSSJ").val(Obj.dKSSJ);
    $("#TB_JSSJ").val(Obj.dJSSJ);

    $("#TB_BM_RQ1").val(Obj.dBM_RQ1);
    $("#TB_BM_RQ2").val(Obj.dBM_RQ2);
    $("#TB_QR_RQ1").val(Obj.dQR_RQ1);
    $("#TB_QR_RQ2").val(Obj.dQR_RQ2);
    $("#TB_CJ_RQ1").val(Obj.dCJ_RQ1);
    $("#TB_CJ_RQ2").val(Obj.dCJ_RQ2);
    $("#TB_HF_RQ1").val(Obj.dHF_RQ1);
    $("#TB_HF_RQ2").val(Obj.dHF_RQ2);
    $("#TB_PS_RQ1").val(Obj.dPS_RQ1);
    $("#TB_PS_RQ2").val(Obj.dPS_RQ2);
    //停用人
    $("#LB_ZZRMC").text(Obj.sZZRMC);
    $("#HF_ZZR").val(Obj.iZZR);
    $("#LB_ZZRQ").text(Obj.dZZRQ);
    //登记人
    $("#LB_DJRMC").text(Obj.sDJRMC);
    $("#HF_DJR").val(Obj.iDJR);
    $("#LB_DJSJ").text(Obj.dDJSJ);

    //审核人
    $("#LB_ZXRMC").text(Obj.sZXRMC);
    $("#HF_ZXR").val(Obj.iZXR);
    $("#LB_ZXRQ").text(Obj.dZXRQ);

}

File: ABCRM/ABCRMProc/CrmProc/KFPT/KFPT_HYHDDY.cs
using System.Text;
using System.Threading.Tasks;
using BF.Pub;
using System.Data;
using System.Data.Common;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

using System.Collections.Generic;
using System;

namespace BF.CrmProc.KFPT
{
    public class KFPT_HYHDDY : DJLR_ZXQDZZ_CLass
    {
        public int iHDID
        {
            set { iJLBH = value; }
            get { return iJLBH; }
        }
        public string dKSSJ = string.Empty;
        public string dJSSJ = string.Empty;
        public string dZZSJ = string.Empty;
        public int iRS = 0;
        public double fXFJE = 0;
        public double fJF = 0;
        public string sHDMC = string.Empty;
        public string sHDNR = string.Empty;
        public string dBM_RQ1 = string.Empty;
        public string dBM_RQ2 = string.Empty;
        public string dQR_RQ1 = string.Empty;
        public string dQR_RQ2 = string.Empty;
        public string dCJ_RQ1 = string.Empty;
        public string dCJ_RQ2 = string.Empty;
        public string dHF_RQ1 = string.Empty;
        public string dHF_RQ2 = string.Empty;
        public string dPS_RQ1 = string.Empty;
        public string dPS_RQ2 = string.Empty;
        public string dZHXFRQ = string.Empty;
        public int iHDStatus = 0;



        public override void DeleteDataQuery(out string msg, CyQuery query, DateTime serverTime)
        {
            CrmLibProc.DeleteDataTables(query, out msg, "HYK_HDNRDEF", "HDID", iJLBH);
        }


        public override List<Object> SearchDataQuery(CyQuery query, DateTime serverTime)
        {
            List<Object> lst = new List<Object>();
            CondDict.Add("iJLBH", "H.HDID");
            CondDict.Add("sSTATUS", "H.STATUS");
            CondDict.Add("dKSRQ", "H.KSSJ");
            CondDict.Add("dJSRQ", "H.JSSJ");
            CondDict.Add("iDJR", "DJR");
            CondDict.Add("sDJRMC", "DJRMC");
            CondDict.Add("dDJSJ", "DJSJ");
            CondDict.Add("iZXR", "ZXR");
            CondDict.Add("sZXRMC", "ZXRMC");
            CondDict.Add("dZXRQ", "ZXRQ");
            CondDict.Add("iZZR", "ZZR");
            CondDict.Add("sZZRMC", "ZZRMC");
            CondDict.Add("dZZRQ", "ZZRQ");
            CondDict.Add("sQZLXMC", "QZLXMC");
            query.SQL.Text = "select H.* from  HYK_HDNRDEF H";
            query.SQL.Add("where H.HDID is not null");
            SetSearchQuery(query, lst);

            return lst;
        }

        public override object SetSearchData(CyQuery query)
        {
            KFPT_HYHDDY item = new KFPT_HYHDDY();
            item.sHDMC = query.FieldByName("HDMC").AsString;
            item.iJLBH = query.FieldByName("HDID").AsInteger;
            item.iRS = query.FieldByName("RS").AsInteger;
            item.fXFJE = query.FieldByName("XFJE").AsFloat;
            item.fJF = query.FieldByName("JF").AsFloat;
            item.sHDNR = query.FieldByName("HDNR").AsString;
            item.dKSSJ = FormatUtils.DateToString(query.FieldByName("KSSJ").AsDateTime);
            item.dJSSJ = FormatUtils.DateToString(query.FieldByName("JSSJ").AsDateTime);
            item.dZHXFRQ = FormatUtils.DateToString(query.FieldByName("ZHXFRQ").AsDateTime);
            item.iDJR = query.FieldByName("DJR").AsInteger;
            item.sDJRMC = query.FieldByName("DJRMC").AsString;
            item.dDJSJ = FormatUtils.DatetimeToString(query.FieldByName("DJSJ").AsDateTime);
            item.iZXR = query.FieldByName("ZXR").AsInteger;
            item.sZXRMC = query.FieldByName("ZXRMC").AsString;
            item.dZXRQ = FormatUtils.DateToString(query.FieldByName("ZXRQ").AsDateTime);
            item.dZZRQ = FormatUtils.DateToString(query.FieldByName("ZZSJ").AsDateTime);
            item.iZZR = query.FieldByName("ZZR").AsInteger;
            item.sZZRMC = query.FieldByName("ZZRMC").AsString;
            item.iSTATUS = query.FieldByName("STATUS").AsInteger;
            item.dBM_RQ1 = FormatUtils.DateToString(query.FieldByName("BM_RQ1").AsDateTime);
            item.dBM_RQ2 = FormatUtils.DateToString(query.FieldByName("BM_RQ2").AsDateTime);
            item.dCJ_RQ1 = FormatUtils.DateToString(query.FieldByName("CJ_RQ1").AsDateTime);
            item.dCJ_RQ2 = FormatUtils.DateToString(query.FieldByName("CJ_RQ2").AsDateTime);
            item.dHF_RQ1 = FormatUtils.DateToString(query.FieldByName("HF_RQ1").AsDateTime);
            item.dHF_RQ2 = FormatUtils.DateToString(query.FieldByName("HF_RQ2").AsDateTime);
            item.dPS_RQ1 = FormatUtils.DateToString(query.FieldByName("PS_RQ1").AsDateTime);
            item.dPS_RQ2 = FormatUtils.DateToString(query.FieldByName("PS_RQ2").AsDateTime);
            item.dQR_RQ1 = FormatUtils.DateToString(query.FieldByName("QR_RQ1").AsDateTime);
            item.dQR_RQ2 = FormatUtils.DateToString(query.FieldByName("QR_RQ2").AsDateTime);
            item.iSTATUS = query.FieldByName("STATUS").AsInteger;
            return item;
        }
        public override void SaveDataQuery(out string msg, CyQuery query, DateTime serverTime)
        {
            msg = string.Empty;
            if (iJLBH != 0)
            {
                DeleteDataQuery(out msg, query, serverTime);
            }
            else
                iJLBH = SeqGenerator.GetSeq("HYK_HDNRDEF");

            query.SQL.Text = "insert into HYK_HDNRDEF(HDID,HDMC,KSSJ,JSSJ,DJR,DJRMC,DJSJ,XFJE,JF,ZHXFRQ,RS,HDNR,STATUS,BM_RQ1,BM_RQ2,QR_RQ1,QR_RQ2,CJ_RQ1,CJ_RQ2,HF_RQ1,HF_RQ2,PS_RQ1,PS_RQ2)";
            query.SQL.Add("values(:HDID,:HDMC,:KSSJ,:JSSJ,:DJR,:DJRMC,:DJSJ,:XFJE,:JF,:ZHXFRQ,:RS,:HDNR,:STATUS,:BM_RQ1,:BM_RQ2,:QR_RQ1,:QR_RQ2,:CJ_RQ1,:CJ_RQ2,:HF_RQ1,:HF_RQ2,:PS_RQ1,:PS_RQ2)");
            query.ParamByName("HDID").AsInteger = iJLBH;
            query.ParamByName("JSSJ").AsDateTime = FormatUtils.ParseDateString(dJSSJ);
            query.ParamByName("KSSJ").AsDateTime = FormatUtils.ParseDateString(dKSSJ);
            query.ParamByName("DJR").AsInteger = iLoginRYID;
            query.ParamByName("DJRMC").AsString = sLoginRYMC;
            query.ParamByName("DJSJ").AsDateTime = serverTime;
            query.ParamByName("HDNR").AsString = sHDNR;
            query.ParamByName("HDMC").AsString = sHDMC;
            query.ParamByName("XFJE").AsFloat = fXFJE;
            query.ParamByName("JF").AsFloat = fJF;
            query.ParamByName("ZHXFRQ").AsDateTime = FormatUtils.ParseDateString(dZHXFRQ);
            query.ParamByName("RS").AsInteger = iRS;
            query.ParamByName("STATUS").AsInteger = 0;
            query.ParamByName("BM_RQ1").AsDateTime = FormatUtils.ParseDateString(dBM_RQ1);
            query.ParamByName("BM_RQ2").AsDateTime = FormatUtils.ParseDateString(dBM_RQ2);
            query.ParamByName("CJ_RQ1").AsDateTime = FormatUtils.ParseDateString(dCJ_RQ1);
            query.ParamByName("CJ_RQ2").AsDateTime = FormatUtils.ParseDateString(dCJ_RQ2);
            query.ParamByName("HF_RQ1").AsDateTime = FormatUtils.ParseDateString(dHF_RQ1);
            query.ParamByName("HF_RQ2").AsDateTime = FormatUtils.ParseDateString(dHF_RQ2);
            query.ParamByName("PS_RQ1").AsDateTime = FormatUtils.ParseDateString(dPS_RQ1);
            query.ParamByName("PS_RQ2").AsDateTime = FormatUtils.ParseDateString(dPS_RQ2);
            query.ParamByName("QR_RQ1").AsDateTime = FormatUtils.ParseDateString(dQR_RQ1);
            query.ParamByName("QR_RQ2").AsDateTime = FormatUtils.ParseDateString(dQR_RQ2);
            query.ExecSQL();


        }

        //审核
        public override void ExecDataQuery(out string msg, CyQuery query, DateTime serverTime)
        {
            msg = string.Empty;
            ExecTable(query, "HYK_HDNRDEF", serverTime, "HDID", "ZXR", "ZXRMC", "ZXRQ", 1);
        }


        //终止
        public override void StopDataQuery(out string msg, CyQuery query, DateTime serverTime)
        {
            msg = string.Empty;
            ExecTable(query, "HYK_HDNRDEF", serverTime, "HDID", "ZZR", "ZZRMC", "ZZRQ", -1);
        }
    }
}
